lib/sequenceserver/blast/job.rb

```diff
@@ -51,32 +52,10 @@ def run
 
         # Capture error.
         status = $CHILD_STATUS.exitstatus
-        case status
-        when 1 # error in query sequence or options; see [1]
-          efile.open
-
-          # Most of the time BLAST+ generates a verbose error message with
-          # details we don't require.  So we parse out the relevant lines.
-          error = efile.each_line do |l|
-            break Regexp.last_match[1] if l.match(ERROR_LINE)
-          end
-
-          # But sometimes BLAST+ returns the exact/ relevant error message.
-          # Trying to parse such messages returns nil, and we use the error
-          # message from BLAST+ as it is.
-          error = efile.rewind && efile.read unless error.is_a? String
-
-          efile.close
-          fail ArgumentError, error
-        when 2, 3, 4, 255 # see [1]
-          efile.open
-          error = efile.read
-          efile.close
-          fail RuntimeError.new(status, error)
-        end
-
-        success!
+        status!(status)
+        success! if status == 0 && !File.zero?(rfile)
```

Comment:
Also check if file exists. While I have never seen IO redirection not create a file, I think it's good to add it for sanity, completeness, and self-documentation.